src/helm/benchmark/scenarios/mtsamples_replicate_scenario.py

```diff
@@ -0,0 +1,145 @@
+import os
+import requests
+import re
+from bs4 import BeautifulSoup
+from typing import List
+from helm.benchmark.scenarios.scenario import (
+    Scenario,
+    Instance,
+    Reference,
+    CORRECT_TAG,
+    TEST_SPLIT,
+    Input,
+    Output,
+)
+from helm.common.general import ensure_directory_exists
+
+
+class MTSamplesReplicateScenario(Scenario):
+    """
+    MTSamples.com is designed to give you access to a big collection of transcribed medical reports.
+    These samples can be used by learning, as well as working medical transcriptionists for their daily
+    transcription needs. We present the model with patient information and request it to generate a corresponding
+    treatment plan.
+
+    Sample Synthetic Prompt:
+    Given various information about a patient, return a reasonable treatment plan for the patient.
+
+    - Extracts `PLAN`, `SUMMARY`, or `FINDINGS` as the reference (PLAN preferred).
+    - Removes `PLAN` from the input text but keeps other sections.
+    - Ignores files that do not contain any of these reference sections.
+    """
+
+    GITHUB_DIR_URL = "https://github.com/raulista1997/benchmarkdata/tree/main/mtsamples_processed"
+    RAW_BASE_URL = "https://raw.githubusercontent.com/raulista1997/benchmarkdata/refs/heads/main/mtsamples_processed/"
+
+    name = "mtsamples_replicate"
+    description = "A dataset of clinical notes where the model is prompted to generate "
+    "the apprioriate treatment plan for this patient."
+    tags = ["medical", "transcription", "plan_generation"]
+
+    def fetch_file_list(self) -> List[str]:
+        """
+        Scrapes the GitHub directory page to get a list of all `.txt` files.
+        """
+        response = requests.get(self.GITHUB_DIR_URL)
+        if response.status_code != 200:
+            raise Exception(f"Failed to fetch file list from GitHub ({self.GITHUB_DIR_URL})")
+
+        soup = BeautifulSoup(response.text, "html.parser")
+        file_links = [
+            link.text
+            for link in soup.find_all(
+                "a", {"href": re.compile(r"/raulista1997/benchmarkdata/blob/main/mtsamples_processed/.*\.txt$")}
+            )
+        ]
+        return file_links
```

Comment:
Use the GitHub API and pin githash (same comments as for `mtsamples_procedures_scenario`)